pkg/unpack/unpacker.go

```diff
@@ -47,48 +48,170 @@ const (
 	labelSnapshotRef = "containerd.io/snapshot.ref"
 )
 
+// Unpacker unpacks images by hooking into the image handler process.
+// Unpacks happen in the backgrounds and waited on to complete.
+type Unpacker interface {
+	Unpack(images.Handler) images.Handler
+	Wait() (Result, error)
+}
+
+// Result returns information about the unpacks which were completed.
+type Result struct {
+	Unpacks int
+}
+
+type unpackerConfig struct {
+	platforms []*Platform
+
+	content content.Store
+
+	limiter               *semaphore.Weighted
+	duplicationSuppressor kmutex.KeyedLocker
+}
+
+type Platform struct {
+	Platform platforms.Matcher
+
+	SnapshotterName string
+	Snapshotter     snapshots.Snapshotter
+	SnapshotOpts    []snapshots.Opt
+
+	Applier   diff.Applier
+	ApplyOpts []diff.ApplyOpt
+}
```

Comment:
This should have comments about the fields and the use-case of `len(unpacker.unpackerConfig.platforms) > 1`.